src/plugins/dataflow/components/dataflow-program.tsx

```diff
@@ -538,21 +544,36 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
   };
 
   private recordCase = () => {
+    const { recordIndex } = this.props;
+    const { programDataRate } = this.props.tileModel; //grab the program Sampling Rate to write TimeQuantized
+    //attributes order  - Time_Quantized as first column | Time_Actual | + # of nodes
     const aCase: ICaseCreation = {};
+    const timeQuantizedKey = "Time_Quantized";
+    const timeActualKey = "Time_Actual";
+
+    const recordTimeQuantized = (recordIndex * programDataRate) / 1000; //in seconds
+    aCase[timeQuantizedKey] = recordTimeQuantized;
+    if (recordIndex === 0) {
+      this.setState({startTimeActual: Date.now()});
+    }
+    const recordTimeActual = (Date.now() - this.state.startTimeActual) / 1000; //in seconds
```

Comment:
Rather than calling `Date.now()` twice in short succession, set it once at the top of the function:
```typescript
  const now = Date.now()
```
and then use it where you need it:
```suggestion
    if (recordIndex === 0) {
      this.startTimeActual = now;
    }
    const recordTimeActual = (now - this.startTimeActual) / 1000; //in seconds
```